packages/flutter_tools/test/general.shard/dart_plugin_test.dart

```diff
@@ -0,0 +1,953 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/dart/package_map.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
+import 'package:flutter_tools/src/flutter_plugins.dart';
+import 'package:flutter_tools/src/globals_null_migrated.dart' as globals;
+import 'package:flutter_tools/src/plugins.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:mockito/mockito.dart';
+import 'package:package_config/package_config.dart';
+import 'package:yaml/yaml.dart';
+
+import '../src/common.dart';
+import '../src/context.dart' hide FakeOperatingSystemUtils;
+
+void main() {
+  group('Dart plugin registrant', () {
+    FileSystem fs;
+    MockFlutterProject flutterProject;
+    MockFlutterManifest flutterManifest;
+
+    setUp(() async {
+      fs = MemoryFileSystem.test();
+
+      flutterProject = MockFlutterProject();
+
+      flutterManifest = MockFlutterManifest();
+      when(flutterManifest.dependencies).thenReturn(<String>{});
+
+      when(flutterProject.manifest).thenReturn(flutterManifest);
+      when(flutterProject.directory).thenReturn(fs.systemTempDirectory.childDirectory('app'));
+
+      when(flutterProject.flutterPluginsFile).thenReturn(flutterProject.directory.childFile('.flutter-plugins'));
+      when(flutterProject.flutterPluginsDependenciesFile).thenReturn(flutterProject.directory.childFile('.flutter-plugins-dependencies'));
+
+      flutterProject.directory.childFile('.packages').createSync(recursive: true);
+    });
+
+    group('resolvePlatformImplementation', () {
+      test('selects implementation from direct dependency', () async {
```

Comment:
Can this be `testWithoutContext` to make sure we're not hitting the real implementation of anything unintentionally? Same question here and below.

Reply:
ditto

Reply:
I think even if these can't use `testWithoutContext`, they should still be using `testUsingContext` to avoid common sources of non-hermeticness. (They might not really need it now, but consider that the next person to come along might add some calls that break assumptions around using `test` without realizing it.)

Reply:
Actually, this can use `testWithoutContext`, so I have changed it accordingly.